nautobot/extras/jobs.py

```diff
@@ -104,25 +105,185 @@ class Meta:
+            )
+
+        self.log_info("Running job")
+
+    def run(self, *args, **kwargs):  # pylint: disable=arguments-differ
+        """
+        Method invoked when this Job is run.
+        """
+        raise NotImplementedError("Jobs must define the run method.")
+
+    def on_success(self, retval, task_id, args, kwargs):
+        """Success handler.
+
+        Run by the worker if the task executes successfully.
+
+        Arguments:
+            retval (Any): The return value of the task.
+            task_id (str): Unique id of the executed task.
+            args (Tuple): Original arguments for the executed task.
+            kwargs (Dict): Original keyword arguments for the executed task.
+
+        Returns:
+            None: The return value of this handler is ignored.
+        """
+
+    def on_retry(self, exc, task_id, args, kwargs, einfo):
+        """Retry handler.
+
+        This is run by the worker when the task is to be retried.
+
+        Arguments:
+            exc (Exception): The exception sent to :meth:`retry`.
+            task_id (str): Unique id of the retried task.
+            args (Tuple): Original arguments for the retried task.
+            kwargs (Dict): Original keyword arguments for the retried task.
+            einfo (~billiard.einfo.ExceptionInfo): Exception information.
+
+        Returns:
+            None: The return value of this handler is ignored.
+        """
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
+        """Error handler.
+
+        This is run by the worker when the task fails.
+
+        Arguments:
+            exc (Exception): The exception raised by the task.
+            task_id (str): Unique id of the failed task.
+            args (Tuple): Original arguments for the task that failed.
+            kwargs (Dict): Original keyword arguments for the task that failed.
+            einfo (~billiard.einfo.ExceptionInfo): Exception information.
+
+        Returns:
+            None: The return value of this handler is ignored.
+        """
+
+    def after_return(self, status, retval, task_id, args, kwargs, einfo):
+        """
+        Handler called after the task returns.
+
+        Parameters
+            status - Current task state.
+            retval - Task return value/exception.
+            task_id - Unique id of the task.
+            args - Original arguments for the task that returned.
+            kwargs - Original keyword arguments for the task that returned.
+
+        Keyword Arguments
+            einfo - ExceptionInfo instance, containing the traceback (if any).
+        """
+
+        # Cleanup FileProxy objects
+        file_fields = list(self._get_file_vars())
+        file_ids = [kwargs[f] for f in file_fields]
+        if file_ids:
+            self.delete_files(*file_ids)
+
+        self.log_info("Job completed")
+        super().after_return(status, retval, task_id, args, kwargs, einfo=einfo)
```

Comment:
https://docs.celeryq.dev/en/stable/userguide/canvas.html#important-notes

> Note
> 
> If you’re using chords with the Redis result backend and also overriding the Task.after_return() method, you need to make sure to call the super method or else the chord callback won’t be applied.
```

Reply:
This is problematic if we are telling authors to overload this if they need to. Let's make sure this is documented. Also, this appears to have dropped off the docstring:

```
        Returns:
            None: The return value of this handler is ignored.
```